setup/gcp/installer/forseti_server_installer.py

```diff
@@ -92,63 +144,149 @@ def deploy(self, deployment_tpl_path, conf_file_path, bucket_name):
-                constants.FirewallRuleAction.ALLOW,
-                ['tcp:22'],
-                constants.FirewallRuleDirection.INGRESS,
-                0,
-                '0.0.0.0/0')
+            # Create firewall rules.
+            self.create_firewall_rules()
 
         return success, deployment_name
 
+    def merge_old_rules(self):
+        """Merge old rules to new rules."""
+        for v1_rule in self.v1_config.rules:
+            new_rule_path = os.path.join(constants.RULES_DIR_PATH,
+                                         v1_rule.file_name)
+            new_rule = files.read_yaml_file_from_local(new_rule_path)
+            field_identifiers = {'rules': ['name', 'rule_id'],
+                                 'default_identifier': 'name',
+                                 'resource': 'type',
+                                 'bindings': 'role',
+                                 'rule_groups': 'group_id'}
+            merge_engine.merge_object(merge_from=v1_rule.data,
+                                      merge_to=new_rule,
+                                      fields_to_ignore=[],
+                                      field_identifiers=field_identifiers)
+            files.write_data_to_yaml_file(new_rule, new_rule_path)
+
+    def create_firewall_rules(self):
+        """Create firewall rules for Forseti server instance."""
+        # Rule to block out all the ingress traffic.
+        gcloud.create_firewall_rule(
+            self.format_firewall_rule_name('forseti-server-deny-all'),
+            [self.gcp_service_account],
+            constants.FirewallRuleAction.DENY,
+            ['icmp', 'udp', 'tcp'],
+            constants.FirewallRuleDirection.INGRESS,
+            1)
+
+        # Rule to open only port tcp:50051 within the
+        # internal network (ip-ranges - 10.128.0.0/9).
+        gcloud.create_firewall_rule(
+            self.format_firewall_rule_name('forseti-server-allow-grpc'),
+            [self.gcp_service_account],
+            constants.FirewallRuleAction.ALLOW,
+            ['tcp:50051'],
+            constants.FirewallRuleDirection.INGRESS,
+            0,
+            '10.128.0.0/9')
+
+        # Create firewall rule to open only port tcp:22 (ssh)
+        # to all the external traffics from the internet.
+        gcloud.create_firewall_rule(
+            self.format_firewall_rule_name(
+                'forseti-server-allow-ssh-external'),
+            [self.gcp_service_account],
+            constants.FirewallRuleAction.ALLOW,
+            ['tcp:22'],
+            constants.FirewallRuleDirection.INGRESS,
+            0,
+            '0.0.0.0/0')
+
+    def generate_forseti_conf(self):
+        """Generate Forseti conf file.
+
+        If self.migrate_from_v1 is True, pull the v1 configuration
+        file and merge it with v2 template.
+
+        Returns:
+            str: Forseti configuration file path.
+        """
+        def config_swap_fields(new_config, old_config):
```

Comment:
Verb first for method namings: ```swap_config_fields```.
Since we are editing this, make the ```old_config``` to be the first arg, and then followed by ```new_config```.

Reply:
Updated.

Reply:
As before, going left to right, it's more natural to think in terms of where we are starting from, to where we are ending.  But it's not as big of a deal as the `merge_from` and 'merge_to`.  So your call.